src/operator/tensor/elemwise_binary_op-inl.h

```diff
@@ -374,6 +374,72 @@ void ElemwiseBinaryOp::CsrCsrOp(mshadow::Stream<cpu> *s,
   }
 }
 
+template<typename OP>
+struct ElemwiseDnsMapKernel {
+  template<typename DType>
+  static void inline Map(int i, const OpReqType req, DType* out, const DType* dns_data,
+                         const nnvm::dim_t num_rows, const nnvm::dim_t num_cols) {
+    if (i < num_rows*num_cols) {
+      KERNEL_ASSIGN(out[i], req, OP::Map(dns_data[i], DType(0.0f)));
+    }
+  }
+};
+
+template<typename OP>
+struct ElemwiseDnsCsrDnsKernel {
+  template<typename DType, typename IType, typename CType>
+  static void inline Map(int i, const OpReqType req, DType* out, DType* dns_data,
+                         const DType* csr_data, const IType* csr_indices, const CType* csr_indptr,
+                         const nnvm::dim_t num_rows, const nnvm::dim_t num_cols) {
+    if (i < num_rows) {
+      for (int j = csr_indptr[i]; j < csr_indptr[i+1]; ++j) {
+        KERNEL_ASSIGN(out[i * num_cols + csr_indices[j]], req,
+                      OP::Map(dns_data[i * num_cols + csr_indices[j]], csr_data[j]));
+      }
+    }
+  }
+};
+
+/*! \brief CSR -op- CSR binary operator for non-canonical NDArray */
```

Comment:
Will change the comment soon, forgot to change it when copying it over from above.